### PR TITLE
Re-ran all notebooks, enhanced gsea

### PR DESCRIPTION
**Pull request by Ecohen1763**

### CODE REVIEW

**Comment by gwaybio** [comment]
Thanks @Ecohen1763 - congrats on filing this PR. Before I can provide a full review, can you convert all your `.ipynb` files into `.py` files? This makes line diffing much easier.

Please use the command (in your terminal):

```bash
jupyter nbconvert --to=script --FilesWriter.build_directory=scripts/nbconverted *.ipynb
```

inside each module (folder) that you have changed

File: 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py
Comment (gwaybio) [inline 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py]: Please make sure your notebooks are run sequentially. Otherwise, there is no way to confirm data/results fidelity across the notebook.

Comment (gwaybio) [reply 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py]: I will review the full PR after you address this item! :)

File: 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py
Comment (gwaybio) [inline 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py]: try to avoid hardcoding. Is there a way to detect what the range to use is without harcoding?

File: 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py
Comment (gwaybio) [inline 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py]: why print this out?

Comment (gwaybio) [reply 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py]: (same can be said of `all_GSEA_results`!)

File: 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py
Comment (gwaybio) [inline 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py]: can you add documentation describing what the column `rank` is?

Also, checkout the function [`pandas.DataFrame.assign()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.assign.html) - I generally prefer it to creating a new column through subsetting.

File: 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py
Comment (gwaybio) [inline 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py]: Why is this label hardcoded? Won't it change each time? Is there a way to not hardcode?

File: 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py
Comment (gwaybio) [inline 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py]: typo in `tight` - did matplotlib not yell at you for this?

**Review by gwaybio — APPROVED** [review]
this looks great to me! Congrats on your first real PR in the lab! 🎉 

I made several comments. Please address them prior to merging, but I do not need to give this another look. Well done!

Also note that you are filing a PR into WayScience:main from your Ecohen:main

image

Typically, it is best practice to file from a separate, feature-specific branch.

Also, best practice is to keep PRs to one feature at a time. Essentially, the best practice would have been to first re-run all notebooks, file a PR for just this, and then file a new PR for enhanced gsea.

One last question for you to think through: Why do the heatmaps look so different?

File: 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py
Comment (gwaybio) [reply 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py]: (PR comments should involve a discussion - it helps the reader to understand decisions that you as the developer made. You as the developer also benefit from multiple eyes on the PR, and, as the developer, you have final say.)

**Comment by gwaybio** [comment]
Do you have any thoughts on the following question?

> One last question for you to think through: Why do the heatmaps look so different?

File: 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py
Comment (Ecohen1763) [reply 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py]: After further thought, I decided to implement the following code to properly label the generated blitz plots with the related latent dimensions:
`col_titles = df.columns.tolist()`
`dim = col_titles[1]`
While it produces the same output as the accumulator strategy, it is not hardcoded and obtains the z value from the column title of each dataframe in `all_signatures`.

File: 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py
Comment (Ecohen1763) [reply 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py]: I think I accidentally hit the spacebar right before commiting, thanks for catching that!

File: 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py
Comment (Ecohen1763) [reply 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py]: I put those there a while ago to see what each output looked like, but will be deleting before merge

File: 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py
Comment (Ecohen1763) [reply 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py]: I changed the code and added a line to the following to extract the number of columns instead of hardcoding a certain range:
`range = signature.shape[1]`
`for col in signature.iloc[:,1:range].columns:`

File: 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py
Comment (Ecohen1763) [reply 3.analysis/scripts/nbconverted/2.gene_set_enrichment_analysis.py]: I was inspired by a video I watched detailing gsea plotting and was initially planning on including the `rank` column as a way to rank the enrichment of each pathway. However, I realized this was unnecessary because unlike the gsea package used in the video, the rank column is not a necessary input for the blitzgsea plot function.

**Comment by Ecohen1763** [comment]
> One last question for you to think through: Why do the heatmaps look so different?

One reason there is such a difference between the updated and previous heatmap is that the previous heatmap only accounted for 10 latent dimensions and the updated heatmap accounts for 100 (which was determined to be the optimal hyperparameter). Additionally, the different colors used by each heatmap add to the visual difference.

**Pull request merged**
